test/functional/tests/failover_standby/test_failover_advanced.py

```diff
@@ -212,4 +785,211 @@ def test_failover_io_long(pyocf_ctx, cls, cleaning_policy):
+            primary_node.cache_drbd_dev,
+            force=True,
+            cache_mode=CacheMode.WB,
+            cache_line_size=cls,
+            cache_id=cache_id,
+        )
+
+        core = primary_node.cache.add_core(primary_node.core_drbd_dev)
+
+    with TestRun.use_dut(primary_node), TestRun.step(f"Set {cleaning_policy} cleaning policy"):
+        primary_node.cache.set_cleaning_policy(cleaning_policy)
+
+    with TestRun.use_dut(primary_node), TestRun.step("Disable sequential cutoff"):
+        primary_node.cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.use_dut(primary_node), TestRun.step(
+        "Create XFS file system on CAS exported object"
+    ):
+        core.create_filesystem(Filesystem.xfs)
+
+    with TestRun.use_dut(primary_node), TestRun.step(f"Mount file system"):
+        mountpoint = "/tmp/standby_io_test_mount_point"
+        TestRun.executor.run(f"rm -rf {mountpoint}")
+        create_directory(path=mountpoint)
+        core.mount(mountpoint)
+
+    file_path = mountpoint + os.path.sep + "fio_file"
+    with TestRun.use_dut(primary_node):
+        fio = (
+            Fio()
+            .create_command()
+            .direct(True)
+            .read_write(ReadWrite.randrw)
+            .bs_split("4k/10:8k/25:16k/25:32k/20:64k/10:128k/5:256k/5")
+            .file_name(file_path)
+            .rand_seed(TestRun.random_seed)
+            .num_jobs(16)
+            .size(core_size * 0.9)  # leave some room for FS metadata
+            .io_size(Size(0, Unit.Byte))
+            .do_verify(True)
+            .set_param("allrandrepeat", 1)
+            .set_flags("refill_buffers")
+        )
+
+    with TestRun.use_dut(primary_node), TestRun.step("Preallocate fio file in pass-through"):
+        primary_node.cache.set_cache_mode(CacheMode.PT, flush=False)
+        # 0 bytes of actual I/O, *not* time based - will just allocate the file
+        fio.time_based(False).run(timedelta(hours=1))
+        primary_node.cache.set_cache_mode(CacheMode.WB, flush=False)
+
+    with TestRun.use_dut(primary_node), TestRun.step("Wait for DRBD synchronization"):
+        primary_node.cache_drbd.wait_for_sync()
+        primary_node.core_drbd.wait_for_sync()
+
+    with TestRun.use_dut(primary_node), TestRun.step(
+        "Run 4h FIO with data verification: random R/W, 16 jobs, filesystem, "
+        "entire primary storage LBA address range, block size split "
+        "4k/10:8k/25:16k/25:32k/20:64k/10:128k/5:256k/5"
+    ):
+        fio.time_based(True).run_time(timedelta(hours=4)).run()
+
+    with TestRun.use_dut(primary_node), TestRun.step("Switch to WO cache mode without flush"):
+        primary_node.cache.set_cache_mode(CacheMode.WO, flush=False)
+
+    with TestRun.use_dut(primary_node), TestRun.step("Calculate checksum of fio test file(s)"):
+        checksum1 = File(file_path).md5sum()
+        TestRun.executor.run(f"md5sum {core.path}").stdout.split()[0]
```

Comment:
Result of this (rather lengthy) operation is not used. Moreover, calculating md5 of drive containing filesystems may cause some discard-related problems.